src/pointer-analysis/dereference.cpp

```diff
@@ -661,15 +665,16 @@ expr2tc dereferencet::build_reference_to(
   if(mode == FREE)
     return expr2tc();
 
+  // Value set tracking emits objects with some cruft built on top of them.
+  value = get_base_object(value);
```

Comment:
Shall we also consider the offset?

Reply:
I'm not sure I fully understand the question. The final offset is computed as a sum of two offsets: the offset (i.e., "o.offset") to the object (e.g., array, struct, union, ...) plus the offset within the object (i.e., "lexical_offset") that is added later on in this method. Is there anything else that we need to take into account?

Reply:
No. I believe that's all. The question was to make sure that we consider both offsets.